specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json

```diff
@@ -0,0 +1,9155 @@
+            "SelfHosted",
+            "Managed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IntegrationRuntimeType",
+            "modelAsString": true
+          }
+        }
+      },
+      "discriminator": "kind"
+    },
+    "IActionResult": {
+      "type": "object"
+    },
+    "IntegrationRuntimeStatus": {
+      "required": [
+        "kind"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "enum": [
+            "SelfHosted",
+            "Managed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IntegrationRuntimeType",
+            "modelAsString": true
+          }
+        }
+      },
+      "discriminator": "kind"
+    },
+    "IntegrationRuntimeAuthKeys": {
+      "type": "object",
+      "properties": {
+        "authKey1": {
+          "type": "string"
+        },
+        "authKey2": {
+          "type": "string"
+        }
+      }
+    },
+    "IntegrationRuntimeRegenerateKeyParameters": {
+      "type": "object",
+      "properties": {
+        "keyName": {
+          "type": "string"
+        }
+      }
+    },
+    "IntegrationRuntimeList": {
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IntegrationRuntime"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        },
+        "count": {
+          "format": "int64",
+          "type": "integer"
+        }
+      }
+    },
+    "EnableInteractiveQueryForIntegrationRuntimeRequest": {
+      "type": "object",
+      "properties": {
+        "autoTerminationMinutes": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "ManagedVirtualNetworkProperties": {
+      "type": "object",
+      "properties": {
+        "vNetId": {
+          "type": "string",
+          "readOnly": true
+        },
+        "alias": {
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ManagedVirtualNetwork": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/ManagedVirtualNetworkProperties"
+            }
+          ]
+        }
+      }
+    },
+    "ConnectionStateProperties": {
+      "type": "object",
+      "properties": {
+        "actionsRequired": {
+          "type": "string",
+          "readOnly": true
+        },
+        "description": {
+          "type": "string",
+          "readOnly": true
+        },
+        "status": {
+          "type": "string",
+          "readOnly": true
+        }
```

Comment:
Should it be enum?

Reply:
Yes, but there is no clarity on what the expected values are even within my change. Several properties could be enums, but DataFactory's swagger did not mark them as enums and hence swashbuckle did not generate enums from Datafactory provided models. This will be a bigger change and I felt it is a bit out of scope for this PR. Let me know. (also explained in the email)